File: Client-Leger/app/src/main/java/com/zouatene/colorimage/model/drawelement/Drawing.kt
package com.zouatene.colorimage.model

import com.squareup.moshi.Json
import com.zouatene.colorimage.model.drawelement.VectorObjectJSON

enum class Privacy(val value: Int) {
    @Json(name = "public")
    PUBLIC(0),

    @Json(name = "protected")
    PROTECTED(1),

    @Json(name = "private")
    PRIVATE(2),
}

class IDrawing(
    val creationTimestamp: Double?,
    val name: String,
    val objects: MutableList<VectorObjectJSON>?,
    val owner: String,
    val password: String? = null,
    val preview: String,
    val privacy: Privacy,
    val avatar: String?,
    val nbCollaborateurs: Int?,
    val version: Int?,
    val versions: Int?,
    val team: String?
)

class DataDrawing(
    val creationTimestamp: Double?,
    val name: String?,
    val objects: MutableList<VectorObjectJSON>?,
    val owner: String?,
    val password: String? = null,
    val preview: String?,
    val privacy: Int?,
    val avatar: String?,
    val nbCollaborateurs: Int?,
    val version: Int?,
    val versions: Int?
)

File: Serveur/interfaces/drawing/drawing.ts
import Document from 'mongoose';
import { Privacy } from '../../Enum/privacy';
import { VectorObject } from './vectorObject';

export interface IDrawingDB extends Document  {
    name                     :   string,
    owner                    :   string,
    password                 :   string,
    team                     :   string | undefined,
    creationTimestamp        :   number,
    privacy                  :   Privacy,
    objects                  :   Array<VectorObject>
    preview                  :   string,
    version                  :   number,
    versions                 :   number,
    z                        :   number
}

export interface IDrawing  {
    name                     :   string,
    owner                    :   string,
    password                 :   string,
    creationTimestamp        :   number,
    privacy                  :   Privacy,
    objects                  :   Array<VectorObject>,
    preview                  :   string,
    team                     :   string | undefined,
    version                  :   number,
    versions                 :   number,
    z                        :   number
}   

export interface version {
    name                      : string,
    versions                  : Array<{ objects : VectorObject[] }>
}